transform/startf/transform.go

```diff
@@ -168,229 +144,194 @@ func ExecScript(ctx context.Context, target *dataset.Dataset, opts ...func(o *Ex
 	resolve.AllowSet = o.AllowSet
 	resolve.AllowLambda = o.AllowLambda
 	resolve.AllowNestedDef = o.AllowNestedDef
+	resolve.LoadBindsGlobally = true
+	resolve.AllowRecursion = true
+	resolve.AllowGlobalReassign = true
 
 	// add error func to starlark environment
 	starlark.Universe["error"] = starlark.NewBuiltin("error", Error)
 	for key, val := range o.Globals {
 		starlark.Universe[key] = val
 	}
 
-	// set transform details
-	target.Transform.Syntax = "starlark"
-	target.Transform.SyntaxVersion = Version
-
-	script := target.Transform.ScriptFile()
-	// "tee" the script reader to avoid losing script data, as starlark.ExecFile
-	// reads, data will be copied to buf, which is re-set to the transform script
-	buf := &bytes.Buffer{}
-	tr := io.TeeReader(script, buf)
-	pipeScript := qfs.NewMemfileReader(script.FileName(), tr)
-
-	t := &transform{
-		ctx:          ctx,
-		dsLoader:     o.DatasetLoader,
-		eventsCh:     o.EventsCh,
-		target:       target,
-		skyqri:       skyqri.NewModule(o.Repo),
-		stderr:       o.ErrWriter,
-		moduleLoader: o.ModuleLoader,
-		changeSet:    o.ChangeSet,
+	thread := &starlark.Thread{Load: o.ModuleLoader}
+
+	r := &StepRunner{
+		config:    target.Transform.Config,
+		secrets:   o.Secrets,
+		dsLoader:  o.DatasetLoader,
+		eventsCh:  o.EventsCh,
+		writer:    o.ErrWriter,
+		thread:    thread,
+		globals:   starlark.StringDict{},
+		changeSet: o.ChangeSet,
 	}
+	r.stards = stards.NewBoundDataset(target, r.onCommit)
 
-	skyCtx := skyctx.NewContext(target.Transform.Config, o.Secrets)
-	thread := &starlark.Thread{Load: t.ModuleLoader}
+	return r
+}
 
-	// execute the transformation
-	t.globals, err = starlark.ExecFile(thread, pipeScript.FileName(), pipeScript, t.locals())
-	if err != nil {
-		if evalErr, ok := err.(*starlark.EvalError); ok {
-			return fmt.Errorf(evalErr.Backtrace())
-		}
-		return err
+// RunStep runs the single transform step using the dataset
+func (r *StepRunner) RunStep(ctx context.Context, ds *dataset.Dataset, st *dataset.TransformStep) error {
+	r.globals["print"] = starlark.NewBuiltin("print", r.print)
+	r.globals["load_version"] = starlark.NewBuiltin("load_version", r.LoadVersionFunc(ctx, ds))
```

Comment:
So `load_version` is staying despite us keeping the name `Dataset` instead of `Version`?